src/code/z_snap.c

```diff
@@ -1,57 +1,66 @@
 #include "global.h"
-
-typedef struct {
-    Actor actor;
-    s32 (*pictoFunc)(PlayState* play, Actor* actor);
-} PictoActor;
-
-s32 func_8013A240(PlayState* play) {
+#include "overlays/actors/ovl_En_Kakasi/z_en_kakasi.h"
+
+/**
+ * Test every loaded actor to see if it is a pictographable `PictoActor`, and if so, run its `validationFunc` to set
+ * appropriate flags.
+ *
+ * @return s32 Number of pictograph actors validly captured.
+ */
+s32 Snap_RecordPictographedActors(PlayState* play) {
     PictoActor* pictoActor;
     Actor* actor;
-    s32 type = 0;
+    s32 category = ACTORCAT_SWITCH;
```

Comment:
When looping through an entire enum from 0 - max, I tend to leave it as 0 and not the enum, because "0" is more important than the value of the enum itself (i.e. shifting `ACTORCAT_SWITCH` to a different value shouldn't shift this value away from 0)

Reply:
I'd be inclined to add `ACTORCAT_MIN` for such situations.

Reply:
Iterating over an enum from 0 to max is ubiquitous over the repo, would you propose a `_MIN` for all those enums?

Reply:
Yes, I've been thinking about this for a while. You'd do the same for different sections of enums that are made out of chunks, like masks in the item enum (or a less bad example...)

Reply:
Would look like
```c
typedef enum {
    /* 0x00 */ ACTORCAT_MIN,
    /* 0x00 */ ACTORCAT_SWITCH = ACTORCAT_MIN,
    /* 0x01 */ ACTORCAT_BG,
    /* 0x02 */ ACTORCAT_PLAYER,
    /* 0x03 */ ACTORCAT_EXPLOSIVES,
    /* 0x04 */ ACTORCAT_NPC,
    /* 0x05 */ ACTORCAT_ENEMY,
    /* 0x06 */ ACTORCAT_PROP,
    /* 0x07 */ ACTORCAT_ITEMACTION,
    /* 0x08 */ ACTORCAT_MISC,
    /* 0x09 */ ACTORCAT_BOSS,
    /* 0x0A */ ACTORCAT_DOOR,
    /* 0x0B */ ACTORCAT_CHEST,
    /* 0x0C */ ACTORCAT_MAX
} ActorType;
```
but I won't do it for now.